### PR TITLE
adding baseline security controls and CI workflow to check compliance

### PR DESCRIPTION
**Pull request by xUnholy**
Signed-off-by: Michael Fornaro <20387402+xUnholy@users.noreply.github.com>

# Description

Creating a compliance workflow and executing conftest against all the resources within the cluster to detect secuirty risks and/or concerns. 

## Type Of Change

To help us figure out who should review this PR, please put an X in all the areas that this PR affects.

- [ ] Docs
- [ ] Installation
- [ ] Performance and Scalability
- [x] Security
- [ ] Test and/or Release
- [ ] User Experience

## Issue Ref (Optional)

Which issue(s) this PR fixes (optional, using fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when the PR gets merged): Fixes #

## Notes

Add special notes for your reviewer here.

### CODE REVIEW

File: policies/P.C.03.rego [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package containers_securitycontext_capabilities_add_index_sys_admim
+
+import data.lib.kubernetes
+
+# https://kubesec.io/basics/containers-securitycontext-capabilities-add-index-sys-admin/
+violation[msg] {
+	kubernetes.containers[container]
+	added_capability(container, "CAP_SYS_ADMIN")
```
Comment (hsy3418) [inline policies/P.C.03.rego:8]: what if there will be multiple capabilities be allowed?

File: policies/P.C.09.rego [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package main
+
+import data.lib.kubernetes
+
+# https://kubesec.io/basics/spec-hostaliases/
+violation[msg] {
+	kubernetes.pods[pod]
+	pod.spec.hostAliases
+	msg = kubernetes.format(sprintf("The %s %s is managing host aliases", [kubernetes.kind, kubernetes.name]))
```
Comment (hsy3418) [inline policies/P.C.09.rego:9]: by looking at this message, not very sure what need to do, is this trying to say the pod shouldn't have a hostAliases specified?

File: policies/P.C.03.rego (hunk 1 above)
Comment (xUnholy) [reply policies/P.C.03.rego:8]: This is only disallowing one singular capability, although I might end up adding more in the future for now the kubesec recommend this one be blocked.

File: policies/P.C.09.rego (hunk 2 above)
Comment (xUnholy) [reply policies/P.C.09.rego:9]: This is a good point, I think at some point I need to standardize the error messaging. I'll raise a ticket for that.

File: policies/P.C.06.rego [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+no_read_only_filesystem(c) {
+	not kubernetes.has_field(c, "securityContext")
+}
+
+no_read_only_filesystem(c) {
+	kubernetes.has_field(c, "securityContext")
+	not kubernetes.has_field(c.securityContext, "readOnlyRootFilesystem")
+}
```
Comment (scpandit) [inline policies/P.C.06.rego:19]:
```suggestion
violation[msg] {
	kubernetes.containers[container]
	not is_read_only_filesystem(container)
	msg = kubernetes.format(sprintf("%s in the %s %s is not using a read only root filesystem", [container.name, kubernetes.kind, kubernetes.name]))
}

is_read_only_filesystem(c) {
	c.securityContext.readOnlyRootFilesystem == true
}
```

File: policies/P.C.15.rego [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+priviledge_escalation_allowed(c) {
+	not kubernetes.has_field(c, "securityContext")
+}
+
+priviledge_escalation_allowed(c) {
+	kubernetes.has_field(c, "securityContext")
+	kubernetes.has_field(c.securityContext, "allowPrivilegeEscalation")
+}
```
Comment (scpandit) [inline policies/P.C.15.rego:18]: In rego/opa we don't have to check if intermediate keys exists in an object.
Meaning, if you want to check if ```obj.key1.key2``` you don't have to check if key1 exists in obj. You can directly check if ```obj.key1.key2 == true`` for example

Comment (scpandit) [reply policies/P.C.15.rego:18]: And I think a request will be violated even if it has ```allowPrivilegeEscalation = false```
So rather be specific and test
```suggestion
priviledge_escalation_allowed(c) {
  	c.securityContext. allowPrivilegeEscalation == true
  }
```

**Pull request merged**